B/Birch/build_tarballs.jl

```diff
@@ -0,0 +1,49 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Birch"
+version = v"1.0.0"
```

Comment:
Same question as for LibBirch: what is the correct way to handle packages such as Birch that consider master to be stable and do not have any releases (and only outdated tags v0 and v1)?

Reply:
We automate a lot of stuff in the Birch CI pipeline. The version number is defined by the command:
```
(git describe --long || echo) | sed -E 's/v([0-9]+)-([0-9]+)-g[0-9a-f]+/\1.\2/'
```
e.g. the current version is 1.75, which is 75 commits after the most recent tag, `v1`. These tags `v0` and `v1` are not outdated, they are just the only manual part of the process. The next major update will be manually tagged `v2`, for example, and then each commit after that will automatically become 2.1, 2.2, 2.3, etc.

How do we intend this Julia package to be updated? Will it do e.g. a nightly build, pulling from the GitHub repo? Or can we do it the other way, pushing updates to the Julia package from the Birch CI pipeline after all tests pass (as we do with Open Build Service, and Homebrew packages)? We could, for example, set up a repo like [homebrew-birch](https://github.com/lawmurray/homebrew-birch) where we push tarballs for the Julia package to use, with correct version numbers, etc. But I'm not familiar with Julia packaging here.

Can chat offline if that helps, and work out a system where we can keep this automated while also providing the right versioning for the Julia package.

Reply:
> We automate a lot of stuff in the Birch CI pipeline. The version number is defined by the command:
> 
> ```
> (git describe --long || echo) | sed -E 's/v([0-9]+)-([0-9]+)-g[0-9a-f]+/\1.\2/'
> ```
> 
> e.g. the current version is 1.75, which is 75 commits after the most recent tag, `v1`. These tags `v0` and `v1` are not outdated, they are just the only manual part of the process. The next major update will be manually tagged `v2`, for example, and then each commit after that will automatically become 2.1, 2.2, 2.3, etc.

Ah great, I didn't know this! I was definitely biased by my experiences with the Julia package ecosystem and therefore looked for Github tags or releases of Birch with new bugfixes or features after v1 had been tagged. bugfix and/or patch releases. But given what you wrote, I guess it would make sense to use 1.75 etc. as version numbers (I'm not completely sure if one has to use 1.75.0 to comply with the [Julia/semver standards](https://juliapackaging.github.io/BinaryBuilder.jl/dev/building/#Version-number) here). The version number could just be defined dynamically based on the provided git commit.

> How do we intend this Julia package to be updated? Will it do e.g. a nightly build, pulling from the GitHub repo? Or can we do it the other way, pushing updates to the Julia package from the Birch CI pipeline after all tests pass (as we do with Open Build Service, and Homebrew packages)? We could, for example, set up a repo like homebrew-birch where we push tarballs for the Julia package to use, with correct version numbers, etc. But I'm not familiar with Julia packaging here.

I guess most packages are updated manually when a new version of the upstream package is released that people want to use. One option here could be, e.g., to add a Github action to the Birch repo that opens PRs to Yggdrasil with updates of the git hashes in the Birch-related Julia build scripts automatically whenever something is pushed to the master branch and tests pass. It is possible to download and rebundle external tarballs in the Julia build scripts but I think this would be suboptimal since then we could not depend on other JLL packages such as boost, LibYAML, etc.

Regarding the names of the packages: the [main restriction or convention is that]((https://juliapackaging.github.io/BinaryBuilder.jl/dev/building/#Name)

> It should be the name of the upstream package, not for example that of a specific library or executable provided by it, even though they may coincide. The case of the name should match that of the upstream package.

Therefore currently I chose the name `Birch` for the compiler, `LibBirch` for the LibBirch library, and `Birch_Standard` for the standard library, based on how these parts are called in their respective README files. I tried to match the case of the names as well, the underscore is a bit arbitrary though.